compiler/aarch64/codegen/OMRMachine.cpp

```diff
@@ -33,15 +33,87 @@
 #include "il/Node_inlines.hpp"
 #include "infra/Assert.hpp"
 
+// Register Association
+#define ARM64_REGISTER_HEAVIEST_WEIGHT               0x0000ffff
+#define ARM64_REGISTER_INITIAL_PRESERVED_WEIGHT      0x00001000
+#define ARM64_REGISTER_ASSOCIATED_WEIGHT             0x00000800
+#define ARM64_REGISTER_PLACEHOLDER_WEIGHT            0x00000100
+#define ARM64_REGISTER_BASIC_WEIGHT                  0x00000080
+
 OMR::ARM64::Machine::Machine(TR::CodeGenerator *cg) :
       OMR::Machine(cg)
    {
    self()->initializeRegisterFile();
+   self()->clearRegisterAssociations();
+   }
+
+/**
+ * @brief Determines if the passed real register is specified
+ *        as the dependency to the passed virtual register
+ *        at the nearest register dependency before the current instruction.
+ *
+ * @param currentInstruction : current instruction
+ * @param realNum            : real register number
+ * @param virtReg            : virtual register
+ * @return true if the real register is specified as the dependency to the virtual register
+ */
+static bool
+boundNext(TR::Instruction * currentInstruction, int32_t realNum, TR::Register * virtReg)
+   {
+   TR::Instruction * cursor = currentInstruction;
+   TR::RealRegister::RegNum realReg = static_cast<TR::RealRegister::RegNum>(realNum);
+   TR::Node * nodeBBStart = NULL;
+
+   while (cursor->getOpCodeValue() != TR::InstOpCode::proc)
+      {
+      TR::RegisterDependencyConditions * conditions;
+      if ((conditions = cursor->getDependencyConditions()) != NULL)
+         {
+         TR::Register * boundReg = conditions->searchPostConditionRegister(realReg);
+         if (boundReg == NULL)
+            {
+            boundReg = conditions->searchPreConditionRegister(realReg);
+            }
+         if (boundReg != NULL)
+            {
+            if (boundReg == virtReg)
```

Comment:
This part can be simply `return (boundReg == virtReg);`.

Reply:
Thanks. I changed it as you suggested.